src/Nethermind/Nethermind.Core/Extensions/CancellationTokenExtensions.cs

```diff
@@ -38,8 +38,11 @@ public static void CancelDisposeAndClear(ref CancellationTokenSource? cancellati
                 {
                     source.Cancel();
                     source.Dispose();
+                    return true;
```

Comment:
A nitpick: the `CompareExchange` above can be replaced with `Exchange` as you always get the previous value.

Reply:
Well depends, if someone would have swapped it for something else by then, then we could break that logic.